fritz

```diff
@@ -172,6 +200,43 @@ def lint(args):
     subprocess.run(["pre-commit", "run", "--all-files"])
 
 
+def prune(args):
+    """
+    Prune fritz'es docker containers and volumes and reset configs to defaults
+    """
+    go = input(f"Do you want to prune fritz'es docker containers and volumes and deinit submodules? [y/N] ")
+
+    if go.lower() == 'y':
+        # try stopping anything that's running first:
+        stop(args)
+
+        # remove docker images
+        for image_name in ('kowalski_api', 'kowalski_ingester', 'skyportal/web'):
+            p1 = subprocess.Popen(['docker', 'images'], stdout=subprocess.PIPE)
+            p2 = subprocess.Popen(['grep', image_name], stdin=p1.stdout, stdout=subprocess.PIPE)
+            image_id = subprocess.check_output(['awk', '{print $3}'], stdin=p2.stdout, universal_newlines=True).strip()
```

Comment:
Heh, I read this and thought: "isn't this why we have Python?"  But, looking at the Python version I think the Unix pipes version actually reads more clearly :man_shrugging: 

```python
image_lines = subprocess.check_output([
    'docker', 'images',
    '--format', '{{.Repository}} {{.ID}}'
]).decode('utf-8').strip().split('\n')

images = [x.split() for x in image_lines]
removed_ids = [image_id for (repo, image_id) in images if image_name in repo_id]
```

Reply:
Exactly. I thought that since we have to use subprocess to ask docker anyways, so why not just do the whole thing with the Unix pipes.